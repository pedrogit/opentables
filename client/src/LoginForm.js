import React from "react";
import TextField from '@mui/material/TextField';
import Stack from "@mui/material/Stack";
import Button from "@mui/material/Button";
import ButtonGroup from "@mui/material/ButtonGroup";
import Collapse from "@mui/material/Collapse";
import Alert from '@mui/material/Alert';
import AlertTitle from '@mui/material/AlertTitle';
import FormControl from '@mui/material/FormControl';
import FormHelperText from '@mui/material/FormHelperText';

import axios from 'axios';

import VisibilityPasswordTextField from './VisibilityPasswordTextField';
const Errors = require('../../client/src/common/errors');

function LoginForm({isVisible, msg, toggleLogin}) {
  const emailRef = React.useRef();
  const passwordRef = React.useRef();
  const [showInvalidLoginHelper, setShowInvalidLoginHelper] = React.useState(false);
  const [performAction, setPerformAction] = React.useState(true);

  msg = {
    severity: 'info',
    title: '',
    msg: '',
    ...msg
  }

  const handleClose = () => {
    emailRef.current.value = null;
    passwordRef.current.value = null;
    toggleLogin(false, '');
    setPerformAction(true);
    setShowInvalidLoginHelper(false);
  }

  const handleKeyDown = (e) => {
    if(e.keyCode === 13){
      doAction();
    }
  }

  var doAction = function() {
    if (msg && msg.action !== undefined) {

      // do not perform the action on the next change of state (only when click on the ok button)
      setPerformAction(false);
      if (emailRef.current.value || passwordRef.current.value) {
        msg.action = {
          ...msg.action, 
          headers: {
            'authorization': 'Basic ' + Buffer.from(emailRef.current.value +':' + passwordRef.current.value).toString('base64')
          }
        }
      }
      axios({...msg.action, withCredentials: true})
      //axios(msg.action)
      .then(res => {
        if (res.status === 200 || res.statusText.toUpperCase() === 'OK') {
          handleClose();
          msg.action.callback(true, res.data);
          return true;
        }
      })
      .catch(error => {
        if (error.response.data.err === Errors.ErrMsg.InvalidUser || 
            error.response.data.err === Errors.ErrMsg.CouldNotLogin) {
          setShowInvalidLoginHelper(true);
        }
        if (error.response.data.err === Errors.ErrMsg.Forbidden) {
          toggleLogin(true, msg);
          emailRef.current.focus();
        }

        return false; 
      });
    }
    return false;
  };

  // try to perform the action before opening
  if (performAction && doAction()) {
    return null;
  }

  return (
    <Collapse in={isVisible}>
      <FormControl>
        <Stack spacing={2}>
          <Alert severity={msg.severity} color='primary'>
            <AlertTitle>{msg.title}</AlertTitle>
            {msg.msg}
          </Alert>
          <Stack direction="row" spacing={2}>
            <TextField 
              variant="outlined"
              required
              size="small"
              fullWidth
              label="Email Address" 
              inputRef={emailRef}
              onChange={() => setShowInvalidLoginHelper(false)}
              onKeyDown={e => handleKeyDown(e)}
              error={showInvalidLoginHelper}
            />
            <VisibilityPasswordTextField
              variant="outlined"
              size="small"
              required
              fullWidth
              label="Password"
              autoComplete="off"
              inputRef={passwordRef}
              onChange={() => setShowInvalidLoginHelper(false)}
              onKeyDown={e => handleKeyDown(e)}
              error={showInvalidLoginHelper}
            />
            <ButtonGroup variant="contained" size="small">
              <Button onClick={() => handleClose()}>Cancel</Button>
              <Button onClick={() => setPerformAction(true)}>Login</Button>
            </ButtonGroup>
          </Stack>
        </Stack>
        <FormHelperText error={showInvalidLoginHelper}>{showInvalidLoginHelper ? 'Invalid email or password...' : ' '}</FormHelperText>
      </FormControl>
    </Collapse>
  );
}

export default LoginForm;

