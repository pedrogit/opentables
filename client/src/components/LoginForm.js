import React from "react";
import TextField from "@mui/material/TextField";
import Stack from "@mui/material/Stack";
import Button from "@mui/material/Button";
import ButtonGroup from "@mui/material/ButtonGroup";
import Collapse from "@mui/material/Collapse";
import Alert from "@mui/material/Alert";
import AlertTitle from "@mui/material/AlertTitle";
import FormControl from "@mui/material/FormControl";
import FormHelperText from "@mui/material/FormHelperText";
import { useTheme } from "@mui/material/styles";
import axios from "axios";
import Cookies from "js-cookie";
import IconButton from "@mui/material/IconButton";
import LoginIcon from "@mui/icons-material/Login";
import LogoutIcon from "@mui/icons-material/Logout";
import HowToRegIcon from "@mui/icons-material/HowToReg";
import Tooltip from "@mui/material/Tooltip";

import VisibilityPasswordTextField from "./VisibilityPasswordTextField";
import { getUser, getScriptBaseURL } from "../clientUtils";

const Errors = require("../../../common/errors");
const Globals = require("../../../common/globals");

function LoginForm({ authAPIRequest, setAuthAPIRequest, setErrorMsg, sx }) {
  const emailRef = React.useRef();
  const passwordRef = React.useRef();
  const [showInvalidLoginHelper, setShowInvalidLoginHelper] =
    React.useState(false);
  const [loginButtonDisabled, setLoginButtonDisabled] = React.useState(true);
  const [doSuccessCallback, setDoSuccessCallback] = React.useState({
    callit: false,
    data: null,
  });

  const theme = useTheme();

  React.useEffect(() => {
    if (authAPIRequest && !authAPIRequest.tryBeforeShowLogin) {
      // don't make the values disappear when closing
      emailRef.current.value = "";
      passwordRef.current.value = "";
      emailRef.current.focus();
    }
  }, [authAPIRequest]);

  const handleClose = () => {
    setAuthAPIRequest({
      tryBeforeShowLogin: true,
      warningMsg: authAPIRequest.warningMsg,
      callback: authAPIRequest.callback,
    });
    setLoginButtonDisabled(true);
    setShowInvalidLoginHelper(false);
  };

  const handleChange = () => {
    setLoginButtonDisabled(
      !(emailRef.current.value && passwordRef.current.value)
    );
  };

  // handle callback only after the login dialog has closed (so the edit popover gets rendered at the right position)
  const handleSuccessCallback = () => {
    if (doSuccessCallback.callit) {
      if (
        authAPIRequest.callback &&
        typeof authAPIRequest.callback === "function"
      ) {
        authAPIRequest.callback(true, doSuccessCallback.data);
      }
      setDoSuccessCallback({ callit: false, data: null });
    }
  };

  const doAction = (addCredentials = false) => {
    if (
      authAPIRequest !== undefined &&
      authAPIRequest.method &&
      authAPIRequest.method !== null
    ) {
      const scriptURL = getScriptBaseURL();
      axios({
        method: authAPIRequest.method,
        url: `${scriptURL}api/opentables/${authAPIRequest.urlParams}`,
        data: authAPIRequest.data,
        headers: addCredentials && {
          authorization: `Basic ${Buffer.from(
            `${emailRef.current.value}:${passwordRef.current.value}`
          ).toString("base64")}`,
        },
        withCredentials: true,
      })
        .then((res) => {
          if (
            res.status === 200 ||
            res.status === 201 ||
            res.statusText.toUpperCase() === "OK"
          ) {
            handleClose();
            if (!authAPIRequest.tryBeforeShowLogin && !res.data) {
              setDoSuccessCallback({ callit: true, data: null });
            } else if (
              authAPIRequest.callback &&
              typeof authAPIRequest.callback === "function"
            ) {
              authAPIRequest.callback(true, res.data);
            }
            return true;
          }
          return false;
        })
        .catch((error) => {
          if (
            error &&
            error.response &&
            error.response.data &&
            error.response.data.err
          ) {
            // handle bad credentials
            if (
              error.response.data.err === Errors.ErrMsg.InvalidUser ||
              error.response.data.err === Errors.ErrMsg.CouldNotLogin
            ) {
              setShowInvalidLoginHelper(true);
            }
            // good credential but still forbidden (leave login form open)
            else if (error.response.status === 403) {
              setAuthAPIRequest({
                ...authAPIRequest,
                tryBeforeShowLogin: false,
                displayCount:
                  authAPIRequest.displayCount === undefined
                    ? 0
                    : authAPIRequest.displayCount + 1,
              });
            }
            // other server errors
            else {
              setAuthAPIRequest({
                tryBeforeShowLogin: false,
              });
              setErrorMsg({ text: error.response.data.err });
              if (
                authAPIRequest.callback &&
                typeof authAPIRequest.callback === "function"
              ) {
                authAPIRequest.callback(false, error.response.data.err);
              }
            }
          }
          // connection errors
          else {
            setAuthAPIRequest({
              tryBeforeShowLogin: false,
            });
            setErrorMsg({ text: error.message });
            if (
              authAPIRequest.callback &&
              typeof authAPIRequest.callback === "function"
            ) {
              authAPIRequest.callback(false, error.message);
            }
          }
          return false;
        });
    }
    return false;
  };

  const handleKeyDown = (e) => {
    setShowInvalidLoginHelper(false);
    if (e.keyCode === 13) {
      doAction(true);
    }
  };

  // try to perform the action before opening
  if (
    authAPIRequest !== undefined &&
    authAPIRequest.tryBeforeShowLogin !== undefined &&
    authAPIRequest.tryBeforeShowLogin === true
  ) {
    doAction();
  }

  return (
    <Stack
      id="loginForm"
      sx={{ backgroundColor: theme.palette.primary.palebg }}
    >
      <Collapse
        in={
          authAPIRequest &&
          !authAPIRequest.tryBeforeShowLogin &&
          authAPIRequest.method &&
          authAPIRequest.method !== null
        }
        onExited={handleSuccessCallback}
      >
        <Stack sx={{ ...sx }}>
          <FormControl sx={{ width: "100%" }}>
            <Stack spacing={2} padding="5px">
              <Alert
                severity={authAPIRequest.warningMsg ? "warning" : "info"}
                color="primary"
                sx={{
                  padding: 0,
                  backgroundColor: theme.palette.primary.palebg,
                }}
              >
                <AlertTitle>
                  {(authAPIRequest.warningMsg
                    ? Globals.permissionDenied
                    : "Login") +
                    (authAPIRequest.displayCount &&
                    authAPIRequest.displayCount > 1
                      ? ` (${authAPIRequest.displayCount})`
                      : "")}
                </AlertTitle>
                {
                  `${
                    authAPIRequest.warningMsg
                      ? `You do not have permission to ${authAPIRequest.warningMsg}. `
                      : ""
                  }Please login with valid credentials...`
                  // authAPIRequest.msg === undefined || authAPIRequest.msg.text === undefined ? '' : authAPIRequest.msg.text
                }
              </Alert>

              <Stack direction={{ xs: "column", sm: "row" }} spacing={2}>
                <TextField
                  variant="outlined"
                  required
                  size="small"
                  fullWidth
                  label="Email Address"
                  inputRef={emailRef}
                  onChange={() => handleChange()}
                  onKeyDown={(e) => handleKeyDown(e)}
                  error={showInvalidLoginHelper}
                  InputProps={{
                    id: "emailInput",
                    sx: { backgroundColor: "white" },
                  }}
                  InputLabelProps={{ shrink: true }}
                />
                <VisibilityPasswordTextField
                  variant="outlined"
                  size="small"
                  required
                  fullWidth
                  label="Password"
                  autoComplete="off"
                  inputRef={passwordRef}
                  onChange={() => handleChange()}
                  onKeyDown={(e) => handleKeyDown(e)}
                  error={showInvalidLoginHelper}
                  InputLabelProps={{ shrink: true }}
                />
                <Stack direction="row" justifyContent="flex-end">
                  <ButtonGroup variant="contained" size="small">
                    <Button
                      id="loginCancelButton"
                      onClick={() => handleClose()}
                    >
                      Cancel
                    </Button>
                    <Button
                      id="loginButton"
                      onClick={() => doAction(true)}
                      disabled={loginButtonDisabled}
                    >
                      Login
                    </Button>
                  </ButtonGroup>
                </Stack>
              </Stack>
              <FormHelperText
                id="loginHelper"
                error={showInvalidLoginHelper}
                sx={{ mt: "2px", fontSize: "14px", fontStyle: "italic" }}
              >
                {showInvalidLoginHelper
                  ? Errors.ErrMsg.InvalidEmailPassword
                  : " "}
              </FormHelperText>
            </Stack>
          </FormControl>
        </Stack>
      </Collapse>
    </Stack>
  );
}

function LoginButton({ setViewId, setAuthAPIRequest, handleReload, buttons }) {
  const theme = useTheme();

  let loginButtonText = "Login";
  const user = getUser();
  console.log(`Logged ${user}`);
  if (user && user !== Globals.allUserName) {
    loginButtonText = `Logout ${user}`;
  }

  const handleLoginLogout = () => {
    if (user && user !== Globals.allUserName) {
      Cookies.remove("authtoken");
      handleReload();
    } else {
      setAuthAPIRequest({
        method: "get",
        tryBeforeShowLogin: false,
        urlParams: "login",
        callback: (success) => {
          if (success) {
            handleReload(true);
          }
        },
      });
    }
  };

  let loginButton = (
    <Button
      id="loginLogoutButton"
      variant="text"
      color="inherit"
      sx={{ p: theme.openTable.buttonPadding, lineHeight: "normal" }}
      onClick={handleLoginLogout}
    >
      {loginButtonText}
    </Button>
  );

  let signUpButton = (
    <Button
      id="signUpButton"
      variant="text"
      color="inherit"
      sx={{ p: theme.openTable.buttonPadding, lineHeight: "normal" }}
      onClick={() => {
        setViewId(Globals.signUpViewOnUserListViewId);
      }}
    >
      Sign&nbsp;Up
    </Button>
  );

  if (buttons) {
    loginButton = (
      <IconButton
        id="loginLogoutButton"
        aria-label="home"
        color="inherit"
        sx={{ p: theme.openTable.buttonPadding }}
        onClick={handleLoginLogout}
      >
        {loginButtonText === "Login" ? (
          <Tooltip title="Login">
            <LoginIcon fontSize="small" />
          </Tooltip>
        ) : (
          <Tooltip title="Logout">
            <LogoutIcon fontSize="small" />
          </Tooltip>
        )}
      </IconButton>
    );

    signUpButton = (
      <Tooltip title="Sign Up">
        <IconButton
          id="signUpButton"
          aria-label="home"
          color="inherit"
          sx={{ p: theme.openTable.buttonPadding }}
          onClick={() => {
            setViewId(Globals.signUpViewOnUserListViewId);
          }}
        >
          <HowToRegIcon fontSize="small" />
        </IconButton>
      </Tooltip>
    );
  }

  return (
    <>
      {loginButton}
      {loginButtonText === "Login" && signUpButton}
    </>
  );
}

export { LoginForm, LoginButton };
