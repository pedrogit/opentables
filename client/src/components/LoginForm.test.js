/* eslint-env mocha */
import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";

import { LoginForm } from "./LoginForm";

describe("LoginForm", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<LoginForm />, div);
  });

  it("has a valid snapshot", () => {
    const component = renderer.create(<LoginForm />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
