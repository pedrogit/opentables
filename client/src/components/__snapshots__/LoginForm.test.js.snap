// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm has a valid snapshot 1`] = `
<div
  className="css-nen11g-MuiStack-root"
  id="loginForm"
>
  <div
    className="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-bz4dnt-MuiCollapse-root"
    style={
      Object {
        "minHeight": "0px",
      }
    }
  >
    <div
      className="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
    >
      <div
        className="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
      >
        <div
          className="css-nen11g-MuiStack-root"
        >
          <div
            className="MuiFormControl-root css-q8hpuo-MuiFormControl-root"
          >
            <div
              className="css-t070a8-MuiStack-root"
            >
              <div
                className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiAlert-root MuiAlert-standardPrimary MuiAlert-standard css-hp9f8j-MuiPaper-root-MuiAlert-root"
                role="alert"
              >
                <div
                  className="MuiAlert-icon css-1ytlwq5-MuiAlert-icon"
                >
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                    data-testid="InfoOutlinedIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M11,9H13V7H11M12,20C7.59,20 4,16.41 4,12C4,7.59 7.59,4 12,4C16.41,4 20,7.59 20, 12C20,16.41 16.41,20 12,20M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10, 10 0 0,0 12,2M11,17H13V11H11V17Z"
                    />
                  </svg>
                </div>
                <div
                  className="MuiAlert-message css-acap47-MuiAlert-message"
                >
                  <div
                    className="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom MuiAlertTitle-root css-yslrua-MuiTypography-root-MuiAlertTitle-root"
                  >
                    
                  </div>
                  
                </div>
              </div>
              <div
                className="css-1wbgkc2-MuiStack-root"
              >
                <div
                  className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
                  onKeyDown={[Function]}
                >
                  <label
                    className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeSmall MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-1kty9di-MuiFormLabel-root-MuiInputLabel-root"
                    data-shrink={true}
                  >
                    Email Address
                    <span
                      aria-hidden={true}
                      className="MuiInputLabel-asterisk MuiFormLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
                    >
                       
                      *
                    </span>
                  </label>
                  <div
                    className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall css-28zdli-MuiInputBase-root-MuiOutlinedInput-root"
                    onClick={[Function]}
                  >
                    <input
                      aria-invalid={false}
                      autoFocus={false}
                      className="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
                      disabled={false}
                      id="emailInput"
                      onAnimationStart={[Function]}
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      required={true}
                      type="text"
                    />
                    <fieldset
                      aria-hidden={true}
                      className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        className="css-1z7n62"
                      >
                        <span>
                          Email Address
                           *
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
                <div
                  className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
                  onKeyDown={[Function]}
                >
                  <label
                    className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeSmall MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-1kty9di-MuiFormLabel-root-MuiInputLabel-root"
                    data-shrink={true}
                  >
                    Password
                    <span
                      aria-hidden={true}
                      className="MuiInputLabel-asterisk MuiFormLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
                    >
                       
                      *
                    </span>
                  </label>
                  <div
                    className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall MuiInputBase-adornedEnd css-owce68-MuiInputBase-root-MuiOutlinedInput-root"
                    onClick={[Function]}
                  >
                    <input
                      aria-invalid={false}
                      autoComplete="off"
                      autoFocus={false}
                      className="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputSizeSmall MuiInputBase-inputAdornedEnd css-19qh8xo-MuiInputBase-input-MuiOutlinedInput-input"
                      disabled={false}
                      id="passwordInput"
                      onAnimationStart={[Function]}
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      required={true}
                      type="password"
                    />
                    <div
                      className="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeSmall css-1laqsz7-MuiInputAdornment-root"
                    >
                      <button
                        aria-label="Toggle password visibility"
                        aria-labelledby={null}
                        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                        data-mui-internal-clone-element={true}
                        disabled={false}
                        onBlur={[Function]}
                        onClick={[Function]}
                        onContextMenu={[Function]}
                        onDragLeave={[Function]}
                        onFocus={[Function]}
                        onKeyDown={[Function]}
                        onKeyUp={[Function]}
                        onMouseDown={[Function]}
                        onMouseLeave={[Function]}
                        onMouseOver={[Function]}
                        onMouseUp={[Function]}
                        onTouchEnd={[Function]}
                        onTouchMove={[Function]}
                        onTouchStart={[Function]}
                        tabIndex={0}
                        type="button"
                      >
                        <svg
                          aria-hidden={true}
                          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                          data-testid="VisibilityOffIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M12 7c2.76 0 5 2.24 5 5 0 .65-.13 1.26-.36 1.83l2.92 2.92c1.51-1.26 2.7-2.89 3.43-4.75-1.73-4.39-6-7.5-11-7.5-1.4 0-2.74.25-3.98.7l2.16 2.16C10.74 7.13 11.35 7 12 7zM2 4.27l2.28 2.28.46.46C3.08 8.3 1.78 10.02 1 12c1.73 4.39 6 7.5 11 7.5 1.55 0 3.03-.3 4.38-.84l.42.42L19.73 22 21 20.73 3.27 3 2 4.27zM7.53 9.8l1.55 1.55c-.05.21-.08.43-.08.65 0 1.66 1.34 3 3 3 .22 0 .44-.03.65-.08l1.55 1.55c-.67.33-1.41.53-2.2.53-2.76 0-5-2.24-5-5 0-.79.2-1.53.53-2.2zm4.31-.78 3.15 3.15.02-.16c0-1.66-1.34-3-3-3l-.17.01z"
                          />
                        </svg>
                        <span
                          className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                        />
                      </button>
                    </div>
                    <fieldset
                      aria-hidden={true}
                      className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        className="css-1z7n62"
                      >
                        <span>
                          Password
                           *
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
                <div
                  className="css-181c0ax-MuiStack-root"
                >
                  <div
                    className="MuiButtonGroup-root MuiButtonGroup-contained css-zqcytf-MuiButtonGroup-root"
                    role="group"
                  >
                    <button
                      className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButtonBase-root MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContainedPrimary css-11qr2p8-MuiButtonBase-root-MuiButton-root"
                      disabled={false}
                      id="loginCancelButton"
                      onBlur={[Function]}
                      onClick={[Function]}
                      onContextMenu={[Function]}
                      onDragLeave={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      onMouseDown={[Function]}
                      onMouseLeave={[Function]}
                      onMouseUp={[Function]}
                      onTouchEnd={[Function]}
                      onTouchMove={[Function]}
                      onTouchStart={[Function]}
                      tabIndex={0}
                      type="button"
                    >
                      Cancel
                      <span
                        className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </button>
                    <button
                      className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButtonBase-root Mui-disabled MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContainedPrimary css-11qr2p8-MuiButtonBase-root-MuiButton-root"
                      disabled={true}
                      id="loginButton"
                      onBlur={[Function]}
                      onClick={[Function]}
                      onContextMenu={[Function]}
                      onDragLeave={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      onMouseDown={[Function]}
                      onMouseLeave={[Function]}
                      onMouseUp={[Function]}
                      onTouchEnd={[Function]}
                      onTouchMove={[Function]}
                      onTouchStart={[Function]}
                      tabIndex={-1}
                      type="button"
                    >
                      Login
                    </button>
                  </div>
                </div>
              </div>
              <p
                className="MuiFormHelperText-root MuiFormHelperText-sizeMedium MuiFormHelperText-contained css-374tsp-MuiFormHelperText-root"
                id="loginHelper"
              >
                <span
                  className="notranslate"
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "&#8203;",
                    }
                  }
                />
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
